lib/msf/core/exploit/remote/ldap/server.rb

```diff
@@ -71,22 +70,45 @@ def on_send_response(cli, data)
         cli.write(data)
       end
 
+      # Perform the required tasks after pdu object is processed
+      # Override this method in modules to take flow control
+      #
+      def on_processed_pdu(processed_data)
+        pdu_type = processed_data[:pdu_type]
+        case pdu_type
+        when Net::LDAP::PDU::BindRequest
+          processed_data
+        else
+          return
+        end
+      end
+
       #
       # Starts the server
       #
       def start_service
         comm = _determine_server_comm(bindhost)
         self.service = Rex::ServiceManager.start(
           Rex::Proto::LDAP::Server,
-          bindhost,
-          bindport,
-          datastore['LdapServerUdp'],
-          datastore['LdapServerTcp'],
-          read_ldif,
-          comm,
+          {
+            bindhost: bindhost,
+            bindport: bindport,
+            udp: datastore['LdapServerUdp'],
+            tcp: datastore['LdapServerTcp'],
+            ldif: read_ldif,
+            comm: comm,
+            challenge: datastore['CHALLENGE'],
+            domain: datastore['Domain'],
+            server: datastore['Server'],
+            dnsname: datastore['DnsName'],
+            dnsdomain: datastore['DnsDomain']
+          },
           { 'Msf' => framework, 'MsfExploit' => self }
         )
 
+        service.processed_pdu_handler do |processed_data|
```

Comment:
This should be a scoped proc if this pattern for registration is going to be held, while I believe either syntax may be valid it would be best be explicit.

```suggestion
        service.processed_pdu_handler = proc do |processed_data|
```

Reply:
As noted in a later comment this could be handle in various ways and is open to discussion.